### PR TITLE
NEOS-1417: update yup error messages

### PR DESCRIPTION
**Pull request by evisdrenova**

### CODE REVIEW

File: frontend/apps/web/app/(mgmt)/[account]/new/job/job-form-validations.ts [hunk 1]
```diff
@@ -129,12 +140,14 @@ export type DefineFormValues = Yup.InferType<typeof DefineFormValues>;
 
 export const ConnectFormValues = SourceFormValues.concat(
   Yup.object({
-    destinations: Yup.array(NewDestinationFormValues).required(),
+    destinations: Yup.array(NewDestinationFormValues).required(
+      'At least one destination Connection is required'
```
Comment (nickzelei) [inline frontend/apps/web/app/(mgmt)/[account]/new/job/job-form-validations.ts:144]: fwiw the `required` here simply says that the `destinations` property is required, not that it has a minimum length (we arent really concerned with that here). In this case it just updates the type to remove the undefined. 

This is generally true for most of the `required` functions (meaning: it's mostly for us to have nice types as we typically have defaults for everything due to how that interacts with react forms)

File: frontend/apps/web/app/api/webhooks/keycloak/slack/route.ts [hunk 2]
```diff
@@ -62,18 +62,20 @@ Example register event for google
   }
 */
 const RegisterEvent = Yup.object({
```
Comment (nickzelei) [inline frontend/apps/web/app/api/webhooks/keycloak/slack/route.ts:64]: A little overkill for webhooks, but it's appreciated nonetheless 😅

File: frontend/apps/web/yup-validations/connections.ts [hunk 3]
```diff
@@ -76,23 +76,30 @@ const connectionNameSchema = Yup.string()
 export const SshTunnelFormValues = Yup.object({
   host: Yup.string(),
   port: Yup.number()
-    .min(0)
-    .when('host', (host, schema) => ([host] ? schema.required() : schema)),
+    .min(0, 'The Port must be greater than 0.')
```
Comment (nickzelei) [inline frontend/apps/web/yup-validations/connections.ts:79]: not sure this message is quite right, we allow the minimum to be 0. So it should be greater than or equal to.

**Review by nickzelei — APPROVED** [review]
LGTM! Left some notes, and one message that should be greater than or equal to instead of just greater than.

File: frontend/apps/web/app/(mgmt)/[account]/new/job/job-form-validations.ts (hunk 1 above)
Comment (evisdrenova) [reply frontend/apps/web/app/(mgmt)/[account]/new/job/job-form-validations.ts:144]: yeah i was kinda going overkill to just set it

**Pull request merged**